### ISSUE TITLE
Support for Craft 4

### ISSUE THREAD

**Issue by romainpoirier** [miranj/craft-obfuscator#6]
When will you support and test the plugin for Craft 4?
https://craftcms.com/docs/4.x/extend/updating-plugins.html
Thank you!

**Comment by luke-nehemedia** [comment]
Would interest me as well, whether you plan to update your plugin for Craft4 or not.

**Comment by rungta** [comment]
@romainpoirier, @luke-nehemedia – We definitely plan on updating all our plugins for Craft 4, starting with this one. Unfortunately I cannot commit a date just yet. But will let you know here as soon as we have some visibility.

**Comment by luke-nehemedia** [comment]
Thank you for your answer. 
I have a similar plugin on the craft pluginstore.  However, since we have more or less the same functionality, I will probably stop developing mine and refer people to yours.

**Comment by anbod-io** [comment]
Just a quick me-too! post on this.

**Comment by swissmexxa** [comment]
+1

**Comment by engram-design** [comment]
Just to chime in, we've added this to our [Craft 3 migration project](https://verbb.io/craft-3) as a few people have requested it.

If you're looking for a new home for Obfuscator, we'd be happy to continue development at Verbb (including updating to Craft 4), but as you've mentioned you plan to upgrade it at some point, that might not be applicable @rungta

**Comment by rungta** [comment]
Thank you for the kind offer @engram-design.

We’re finally embarking on the Craft 4 migrations for our plugins. We just pushed out an update with Craft 4 support for [Contact Form Tuner](https://github.com/miranj/craft-contact-form-tuner) yesterday and will be tackling this plugin next. Hoping to have a release out this week.

**Comment by engram-design** [comment]
Nice one @rungta just figured I'd put the offer out there, I'll leave this with you!

**Comment by rungta** [comment]
Obfuscator 1.1.0 is out now with Craft 4 support.

**Comment by swissmexxa** [comment]
🥳 thank you very much

**Comment by romainpoirier** [comment]
Thank you for the update.
You should update the doc, as from Craft 4: _The {% filter %} tag is removed. Use the {% apply %} tag instead_.

Also, after trying something like this:

```
{% apply enkode_emails %}
	{% set attr = {
		href: 'mailto:{email}' | t({
			email: entry.email
		})
	} %}
	<a {{ attr(attr) }}>
		{{ entry.email }}
	</a>
{% endapply %}
```

I can see this in the HTML source code:
```
<a href="mailto:info@test.com">
    <span id="enkoder_0_1600286958">email hidden; JavaScript is required</span>
    <script id="script_enkoder_0_1600286958" type="text/javascript">
    /* <!-- */
    function hivelogic_enkoder_0_1600286958() {
        var kode = "kode=\"110 114 103 104 64 37 52 52 51 35 52 52 55 35 52 51 54 35 52 51 55 35 57 55 35 54 58 35 52 52 55 35 52 51 54 35 52 53 51 35 52 51 53 35 52 51 55 35 52 52 53 35 52 52 60 35 52 52 54 35 52 51 57 35 55 60 35 52 52 60 35 52 51 55 35 52 52 52 35 58 53 35 52 52 53 35 52 51 55 35 52 52 54 35 52 51 55 35 57 60 35 52 52 60 35 58 57 35 52 53 55 35 55 54 35 52 51 54 35 58 53 35 60 56 35 55 53 35 58 59 35 59 52 35 58 52 35 59 53 35 59 56 35 58 53 35 58 57 35 60 59 35 60 56 35 55 53 35 58 52 35 55 60 35 55 55 35 52 53 51 35 52 52 55 35 52 51 55 35 52 52 60 35 58 56 35 52 52 58 35 59 51 35 59 58 35 57 55 35 58 60 35 52 51 55 35 52 51 54 35 52 52 55 35 52 51 53 35 52 51 55 35 52 51 54 35 59 56 35 59 59 35 58 51 35 58 57 35 52 52 53 35 52 52 55 35 52 52 55 35 52 52 56 35 52 51 55 35 52 52 54 35 52 52 60 35 52 52 54 35 52 51 55 35 55 54 35 52 51 53 35 52 52 59 35 52 52 56 35 52 51 51 35 55 54 35 52 51 55 35 60 56 35 60 56 35 60 56 35 54 58 35 52 51 51 35 57 54 35 52 51 53 35 54 56 35 52 51 51 35 52 52 52 35 52 52 59 35 52 52 59 35 60 56 35 60 56 35 57 55 35 52 51 55 35 60 56 35 54 58 35 52 52 51 35 52 52 54 35 52 51 54 35 52 52 55 35 52 51 54 35 52 51 55 35 52 52 53 35 55 59 35 52 51 59 35 52 51 51 35 52 52 60 35 52 52 52 35 60 56 35 60 56 35 52 52 55 35 54 56 35 60 56 35 54 58 35 52 52 58 35 52 51 58 35 52 51 56 35 52 51 55 35 60 56 35 60 56 35 57 55 35 52 52 53 35 60 56 35 54 58 35 52 51 59 35 52 51 51 35 52 52 60 35 52 52 52 35 57 52 35 52 52 55 35 52 52 54 35 52 51 59 35 52 52 55 35 52 51 56 35 52 51 52 35 57 58 35 52 53 56 35 52 52 55 35 52 52 58 35 52 51 51 35 52 51 52 35 55 60 35 60 56 35 60 56 35 52 51 55 35 60 56 35 60 56 35 60 56 35 54 58 35 52 51 59 35 57 56 35 52 51 56 35 52 52 54 35 57 58 35 52 52 55 35 52 52 55 35 52 51 52 35 52 51 51 35 52 53 56 35 55 60 35 52 52 58 35 52 51 55 35 52 51 52 35 57 54 35 60 56 35 60 56 35 52 51 51 35 56 51 35 57 56 35 60 56 35 60 56 35 55 55 35 60 56 35 54 58 35 57 53 35 55 55 35 54 58 35 57 53 35 52 53 54 35 57 55 35 55 53 35 55 53 35 57 53 35 52 51 56 35 52 52 55 35 52 52 58 35 55 54 35 52 51 59 35 57 55 35 56 52 35 57 53 35 52 51 59 35 57 54 35 55 54 35 52 52 51 35 52 52 55 35 52 51 54 35 52 51 55 35 55 60 35 52 52 52 35 52 51 55 35 52 52 54 35 52 51 57 35 52 52 60 35 52 51 58 35 55 59 35 56 53 35 55 55 35 57 53 35 52 51 59 35 55 57 35 57 55 35 56 54 35 55 55 35 52 53 57 35 52 53 54 35 55 57 35 57 55 35 52 52 51 35 52 52 55 35 52 51 54 35 52 51 55 35 55 60 35 52 51 53 35 52 51 58 35 52 51 51 35 52 52 58 35 57 59 35 52 52 60 35 55 54 35 52 51 59 35 55 57 35 56 53 35 55 55 35 55 57 35 52 52 51 35 52 52 55 35 52 51 54 35 52 51 55 35 55 60 35 52 51 53 35 52 51 58 35 52 51 51 35 52 52 58 35 57 59 35 52 52 60 35 55 54 35 52 51 59 35 55 55 35 52 53 59 35 52 52 51 35 52 52 55 35 52 51 54 35 52 51 55 35 57 55 35 52 53 54 35 55 57 35 55 54 35 52 51 59 35 57 54 35 52 52 51 35 52 52 55 35 52 51 54 35 52 51 55 35 55 60 35 52 52 52 35 52 51 55 35 52 52 54 35 52 51 57 35 52 52 60 35 52 51 58 35 57 57 35 52 52 51 35 52 52 55 35 52 51 54 35 52 51 55 35 55 60 35 52 51 53 35 52 51 58 35 52 51 51 35 52 52 58 35 57 59 35 52 52 60 35 55 54 35 52 52 51 35 52 52 55 35 52 51 54 35 52 51 55 35 55 60 35 52 52 52 35 52 51 55 35 52 52 54 35 52 51 57 35 52 52 60 35 52 51 58 35 55 59 35 56 53 35 55 55 35 57 52 35 55 53 35 55 53 35 55 55 35 57 53 37 62 110 114 103 104 64 110 114 103 104 49 118 115 111 108 119 43 42 35 42 44 62 123 64 42 42 62 105 114 117 43 108 64 51 62 108 63 110 114 103 104 49 111 104 113 106 119 107 62 108 46 46 44 126 123 46 64 86 119 117 108 113 106 49 105 117 114 112 70 107 100 117 70 114 103 104 43 115 100 117 118 104 76 113 119 43 110 114 103 104 94 108 96 44 48 54 44 128 110 114 103 104 64 123 62\";kode=kode.split(\' \');x=\'\';for(i=0;i<kode.length;i++){x+=String.fromCharCode(parseInt(kode[i])-3)}kode=x;",
            i, c, x, script = document.currentScript || document.getElementById("script_enkoder_0_1600286958");
        while (kode.indexOf("getElementById('ENKODER_ID')") === -1) { eval(kode) };
        kode = kode.replace('ENKODER_ID', 'enkoder_0_1600286958');
        eval(kode);
        script && script.parentNode.removeChild(script);
    }
    hivelogic_enkoder_0_1600286958();
    /* --> */
    </script>
</a>
```

As I can see `<a href="mailto:info@test.com">` in clear HTML, the email address is not obfuscated.

**Comment by rungta** [comment]
Updated the docs with the `{% apply %}` tag, and fixed the bug (#4) as well. Thanks for reporting @romainpoirier!
